src/main/java/com/elixr/poc/rest/controller/PurchaseDeletionController.java

```diff
@@ -32,19 +32,12 @@ public PurchaseDeletionController(PurchaseService purchaseService) {
      * @return
      */
     @DeleteMapping("/purchase/{purchaseId}")
-    public ResponseEntity<?> deletePurchase(@PathVariable("purchaseId") UUID purchaseId) {
-        DeleteResponse deleteResponse = new DeleteResponse();
+    public ResponseEntity<?> deletePurchase(@PathVariable("purchaseId") UUID purchaseId) throws GlobalException {
+        CommonErrorResponse deleteResponse;
         HttpStatus httpStatus;
-        try {
-            boolean success = purchaseService.deletePurchaseDetails(purchaseId);
-            deleteResponse.setSuccess(success);
-            deleteResponse.setErrorMessages(ApplicationConstants.DELETED);
-            httpStatus = HttpStatus.OK;
-        } catch (NoRecordFoundException e) {
-            deleteResponse.setSuccess(false);
-            deleteResponse.setErrorMessages(e.getMessage());
-            httpStatus = HttpStatus.BAD_REQUEST;
-        }
+        boolean success = purchaseService.deletePurchaseDetails(purchaseId);
+        deleteResponse = CommonErrorResponse.builder().success(success).errorMessage(ApplicationConstants.SUCCESSFULLY_DELETED).build();
```

Comment:
Why is the response object name as ErrorResponse, even if it is a success? Better to rename the class and attributes of the class, or use another success response class if you have

Reply:
Updated has per your suggestion